src/System.Net.Security/tests/FunctionalTests/NegotiateStreamKerberosTest.cs

```diff
@@ -157,6 +164,50 @@ private async Task VerifyAuthentication(NetworkCredential credential)
             }
         }
 
+        [OuterLoop]
+        [ConditionalFact(nameof(IsClientAvailable))]
+        public async Task NegotiateStream_ServerAuthenticationRemote_Success()
+        {
+            string expectedUser = Configuration.Security.NegotiateClientUser;
+            
+            string expectedAuthenticationType = "Kerberos";
+            bool mutuallyAuthenticated = true;
+
+            using (var controlClient = new TcpClient())
+            {
+                string clientName = Configuration.Security.NegotiateClient.Host;
+                int clientPort = Configuration.Security.NegotiateClient.Port;
+                await controlClient.ConnectAsync(clientName, clientPort)
+                    .TimeoutAfter(TimeSpan.FromSeconds(15));
+                var serverStream = controlClient.GetStream();
+
+                using (var serverAuth = new NegotiateStream(serverStream, leaveInnerStreamOpen: false))
+                {
+                    await serverAuth.AuthenticateAsServerAsync(
+                        CredentialCache.DefaultNetworkCredentials,
+                        ProtectionLevel.EncryptAndSign,
+                        TokenImpersonationLevel.Identification)
+                        .TimeoutAfter(TimeSpan.FromSeconds(15));
+
+                    Assert.True(serverAuth.IsAuthenticated, "IsAuthenticated");
+                    Assert.True(serverAuth.IsEncrypted, "IsEncrypted");
+                    Assert.True(serverAuth.IsSigned, "IsSigned");
+                    Assert.Equal(mutuallyAuthenticated, serverAuth.IsMutuallyAuthenticated);
+
+                    Assert.Equal(expectedAuthenticationType, serverAuth.RemoteIdentity.AuthenticationType);
+                    Assert.Equal(expectedUser, serverAuth.RemoteIdentity.Name);
+
+                    // Receive a message from the client.
+                    var message = "Hello from the client.";
+                    using (var reader = new StreamReader(serverAuth))
+                    {
+                        var response = await reader.ReadToEndAsync().TimeoutAfter(TimeSpan.FromSeconds(15));
+                        Assert.Equal(message, response);
+                    }
+                }
+            }
+        }
+
```

Comment:
Can we now enable tests like https://github.com/dotnet/corefx/blob/d542839aa87e4acbcf8743cd0590214a522a30f7/src/System.Net.Security/tests/FunctionalTests/NegotiateStreamStreamToStreamTest.cs#L16 to run everywhere and not just Windows?

Reply:
We will need to add a conditional for this, "IsNtlmInstalled" (which we already have Capability for).  This test uses loopback which doesn't work with Kerberos and would fall back to NTLM.  The test is already checking for that "Assert.Equals(false, IsMutuallyAuthenticated)".

Reply:
I tried enabling them but they're not going to work as is. The NegotiateStream unix client does not support default credentials without an SPN.
https://github.com/dotnet/corefx/blob/0fd3b7e3e87ca7bb7b46cd0b92cbbc3ddda81b82/src/Common/src/System/Net/Security/NegotiateStreamPal.Unix.cs#L259-L262
Adding an SPN resulted in "SPNEGO cannot find mechanisms to negotiate".
I added IsNtlmInstalled, but I'm going to leave these disabled and come back to it.

Reply:
You're right.  Using default credentials doesn't work with NTLM.  I added other tests that use explicit credentials and that works for both Kerberos and NTLM.  

So, it's fine to leave these as-is for now.  I was planning on doing a rewrite of these anyway at some point as part of my other Enterprise focused work.